routes/create-or-update-file.js

```diff
@@ -0,0 +1,65 @@
+var request = require("request");
+
+module.exports = function(config) {
+  return function(req, res) {
+    if(!config.authorized(req, res)) {
```

Comment:
This is probably an anti-pattern in express. I'd declare this as middleware that runs before this route handler.